src/main/java/org/mdbenefits/app/submission/actions/CheckExpeditedSnapEligibility.java

```diff
@@ -21,41 +23,37 @@ public class CheckExpeditedSnapEligibility implements Action {
-                            isMigrantOrSeasonalFarmWorker && isBelowMoneyOnHandThreshhold)
-                            || calculateUtilitiesExpenses(inputData).compareTo(BigDecimal.ZERO) > 0
-                            && isEligibleByIncomeAndCashOnHandLessThanExpenses;
-            submission.getInputData().put("isEligibleForExpeditedSnap", String.valueOf(isEligibleForExpeditedSnap));
+        boolean isApplyingForExpeditedSnap = Boolean.parseBoolean(
+                inputData.getOrDefault("isApplyingForExpeditedSnap", "false").toString());
+
+        ExpenseCalculator expenseCalculator = new ExpenseCalculator(submission);
+
+        BigDecimal moneyOnHandAmount = convertToBigDecimal(
+                inputData.getOrDefault("expeditedMoneyOnHandAmount", "0").toString());
+        BigDecimal householdIncomeAmount = convertToBigDecimal(
+                inputData.getOrDefault("householdIncomeLast30Days", "0").toString());
+        if (!isApplyingForExpeditedSnap) {
+            // they are in the main flow, so we need to use the income they entered there.
+            IncomeCalculator incomeCalculator = new IncomeCalculator(submission);
+            householdIncomeAmount = BigDecimal.valueOf(incomeCalculator.totalFutureEarnedIncome())
+                    .setScale(2, RoundingMode.HALF_UP);
         }
-    }
 
-    private BigDecimal calculateUtilitiesExpenses(Map<String, Object> inputData) {
-        List<String> allExpenses = new java.util.ArrayList<>();
-        if (!isNoneOfAboveSelection(inputData.get("householdHomeExpenses[]"))) {
-            List<String> utilityTypes = (List<String>) inputData.get("householdHomeExpenses[]");
-            utilityTypes.forEach(val -> {
-                String inputFieldName = HomeExpensesType.getEnumByName(val).getInputFieldName();
-                allExpenses.add(inputData.getOrDefault(inputFieldName, "0").toString());
-            });
-        }
-        return allExpenses.stream().map(n -> convertToBigDecimal(n)).reduce(BigDecimal.ZERO, BigDecimal::add);
+        boolean isMigrantOrSeasonalFarmWorker = inputData.getOrDefault("migrantOrSeasonalFarmWorkerInd", "false").toString()
+                .equals("true");
+        boolean isBelowMoneyOnHandThreshhold = inputData.getOrDefault("householdMoneyOnHandLessThan100", "false").toString()
+                .equals("true");
+        boolean isBelowIncomeThreshhold = inputData.getOrDefault("incomeLessThan150", "false").toString().equals("true");
+
+        boolean isEligibleByIncomeAndCashOnHandLessThanExpenses =
+                householdIncomeAmount.add(moneyOnHandAmount).compareTo(expenseCalculator.totalUtilitiesExpenses()) <= 0;
+
+        boolean isEligibleForExpeditedSnap =
+                (isBelowMoneyOnHandThreshhold && isBelowIncomeThreshhold) || (
+                        isMigrantOrSeasonalFarmWorker && isBelowMoneyOnHandThreshhold)
+                        || expenseCalculator.totalUtilitiesExpenses().compareTo(BigDecimal.ZERO) > 0
```

Comment:
should the last part of this statement after the final || be wrapped in parens like the others?